test/test_tensorexpr.py

```diff
@@ -1225,6 +1225,15 @@ def run_where(x, y):
         y = run_where(a, b)
         np.testing.assert_allclose(x.numpy(), y.numpy())
 
+    def test_unused(self):
```

Comment:
We should also add a function test: 

A(i, j) = rand_like()
B = (2 * A - 1) * (2 * A - 2)
assert(B(i, j) >= -0.25)

If both A are inlined as rand, this is going to fail. But with your fix, this is going to pass.